component.json

```diff
@@ -29,127 +29,67 @@
       "help": {
         "description": "Multiple incoming events can be combined into one CSV file with the write CSV action.",
         "link": "/components/csv/actions#write-csv-attachment"
       },
       "fields": {
-        "includeHeaders": {
-          "label": "Include Headers",
-          "required": false,
-          "viewClass": "SelectView",
-          "description": "Default Yes",
-          "model": {
-            "Yes": "Yes",
-            "No": "No"
-          },
+        "uploadToAttachment": {
+          "label": "Upload CSV as file to attachments",
+          "viewClass": "CheckBoxView",
+          "description": "If checked store the generated CSV data as an attachment",
           "prompt": "Include headers? Default Yes."
         },
-        "writer": {
-          "viewClass": "CSVWriteView"
-        }
-      },
-      "metadata": {
-        "in": {
-          "type": "object",
-          "properties": {}
-        },
-        "out": {}
-      }
-    },
-    "write_attachment_from_json": {
-      "main": "./lib/actions/writeFromJson.js",
-      "title": "Write CSV attachment from JSON Object",
-      "help": {
-        "description": "Multiple incoming events can be combined into one CSV file with the write CSV action.",
-        "link": "/components/csv/actions#write-csv-attachment-from-json-object"
-      },
-      "fields": {
-        "includeHeaders": {
-          "label": "Include Headers",
+        "separator": {
+          "viewClass": "TextFieldView",
           "required": false,
-          "viewClass": "SelectView",
-          "description": "Default Yes",
-          "model": {
-            "Yes": "Yes",
-            "No": "No"
-          },
-          "prompt": "Include headers? Default Yes"
+          "label": "Separator",
+          "note": "A single char used to delimit the CSV file. Default to ','"
         },
-        "separator": {
-          "label": "Separators",
-          "required": true,
-          "viewClass": "SelectView",
-          "model": {
-            "comma": "Comma (,)",
-            "semicolon": "Semicolon (;)",
-            "space": "Space ( )",
-            "tab": "Tab (\\t)",
-            "pipe": "Pipe (\u00A6)"
-          },
-          "prompt": "Choose required CSV delimiter"
+        "order": {
+          "viewClass": "TextFieldView",
+          "required": false,
+          "label": "Column Order",
+          "note": "A string delimited with the separator (use same as above) indicating which columns & in what order the columns should appear in the resulting file. If omitted, the column order in the resulting file will not be deterministic."
         }
       },
       "metadata": {
-        "in": {
-          "type": "object",
-          "properties": {
-            "inputObject": {
-              "type": "object",
-              "properties": {}
-            }
-          }
-        },
+        "in": "./lib/schemas/writeFromStream.in.json",
```

Comment:
One further potential improvement that I just thought of:
> What if metadata was dynamic? If there is a provided **Column Order** then build an input for each column (within the `item` object). If there is not provided **Column Order** then leave `item` as an unstructured object.

I don't make sense if we should try to add it to this issue.

Reply:
Can you give me example?

Current implementation looks like this:
Input object: {a: '🙈', b: '🙉', c: '🙊', d: '😂'}
Column Order: 'd, a, f, g'
Outpu object: {csvString: "😂,🙈,,"}

Reply:
I'm saying instead of having static metadata encoded in the `writeFromStream.in.json` file, we have dynamic metadata (like in the [Zendesk Component](https://github.com/elasticio/zendesk-component/blob/master/component.json#L144)) where inside the [`getMetaModel()` function](https://github.com/elasticio/zendesk-component/blob/master/lib/actions/upsertObjectById.js#L61) we have an if statement.

If there is no column order, than the returned in metadata is 
```
{
    "type": "object",
    "properties": {
        "header": {
            "type": "boolean",
            "required": true,
            "title": "Include Headers"
        },
        "items": {
            "type": "object",
            "required": true,
            "title": "Input Object"
        }
    }
}
```
like we have now. However, if there is a column order like `'d, a, f, g'`, then the in metadata should be
```
{
    "type": "object",
    "properties": {
        "header": {
            "type": "boolean",
            "required": true,
            "title": "Include Headers"
        },
        "items": {
            "type": "object",
            "required": true,
            "title": "Input Object"
            "properties": {
              "d": {
                "type": "string",
                "required": true,
                "title": "d"
              },
              "a": {
                "type": "string",
                "required": true,
                "title": "a"
              },
              "f": {
                "type": "string",
                "required": true,
                "title": "f"
              },
              "g": {
                "type": "string",
                "required": true,
                "title": "g"
              }
            }
        }
    }
}
```

Reply:
Done